docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,365 @@
+
+### Use Cases
+
+#### Recovery from permanent quorum loss
+
+Currently identification and recovery from permament quorum loss is done manaually. The current proposal keeps the identification as manual, requiring an human operator to intervene and confirm that there is indeed a permanent quorum loss and automatic recovery is not possible. Once that is established then the authors now wish to automate the recovery which is currently a [multi-step process](https://github.com/gardener/etcd-druid/blob/master/docs/operations/recovery-from-permanent-quorum-loss-in-etcd-cluster.md) and needs to be done carefully. Automation would ensure that we eliminate any errors from an operator.
+
+##### Task Config
+
+We do not need any config for this task. When creating an instance of `EtcdOperatorTask` for this scenario, `.spec.config` will be set to nil (unset).
+
+##### Pre-Conditions
+
+* There should be a quorum loss in a multi-member etcd cluster. For a single-member etcd cluster, invoking this task is unnecessary as the restoration of the single member is automatically handled by the backup-restore process.
+* There should not already be a permanent-quorum-loss-recovery-task running for the same etcd cluster.
+
+#### Trigger on-demand snapshot compaction
+
+`etcd-druid` provides a configurable [etcd-events-threshold](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/02-snapshot-compaction.md#druid-flags). If and when this threshold is breached then only a [snapshot compaction](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/02-snapshot-compaction.md) is triggered for etcd cluster. However, there are scenarios where an ad-hoc snapshot compaction is required.
+
+##### Possible scenarios
+
+* Full snapshot taken via compaction job is better as compared to full snapshot taken directly from etcd cluster as full snapshot taken via snapshot compaction is taken after [defragmentation](https://etcd.io/docs/v3.2/op-guide/maintenance/#defragmentation) called on etcd, hence this full snapshot is much more compacted, hence less in size.
+* Full snapshot triggered on etcd cluster might cause load on running etcd cluster but full snapshot taken by compaction job is developed separately with zero effect on running cluster.
+
+##### Task Config
+
+We do not need any config for this task. When creating an instance of `EtcdOperatorTask` for this scenario, `.spec.config` will be set to nil (unset).
+
+##### Pre-Conditions
+
+* There should not be a `on-demand snapshot-compaction` task already running for the same etcd cluster.
+* Other than that, it has no pre-conditions as `on-demand snapshot-compaction` runs as a separate job in a seprate pod, hence it doesn't depends on health of etcd cluster members or any other conditions.
```

Comment:
Bullet points are meant for specifying pre-conditions but this point is not a pre-condition. You can just add it as a `NOTE` after the pre-conditions are listed.

```suggestion
> NOTE: `on-demand snapshot-compaction` runs as a separate job in a separate pod, hence it doesn't depends on health of etcd cluster members or any other conditions.
```